crates/core/src/transport/packet_data.rs

```diff
@@ -113,51 +128,137 @@ impl<const N: usize> PacketData<N> {
         Ok(Self {
             data: buffer,
             size: buffer_len,
+            data_type : PhantomData,
         })
     }
+}
+impl<const N: usize> PacketData<AssymetricEncrypted, N> {
+    pub(super) fn encrypt_with_pubkey(data: &[u8], remote_key: &TransportPublicKey) -> Self {
```

Comment:
we can take a packet data here too, e.g.:

```
fn <const M: usize<(data: PacketData<M, UnknownEncrypted>,  ..)

Reply:
I think if we're encrypting then we should require that it's PlainText